File: J1_07 - VerticesPerimetroTriangulo/VPTriangulo/src/Ponto.java
public class Ponto
{
   private double x;
   private double y;

   // Construtor
   public Ponto(double x, double y)
   {
      this.x = x;
      this.y = y;
   }

   public double distancia(Ponto p)
   {
      return (double) Math.sqrt(Math.pow(this.x - p.x, 2) + Math.pow(this.y - p.y, 2));
   }
}

File: desc_CPP_01/src/Ponto.cpp
#include "Ponto.hpp"
#include <cmath>


Ponto::Ponto(double _x, double _y) : x(_x), y(_y) {}

double Ponto::distancia(const Ponto& ponto) const {
    double d = sqrt(pow(ponto.x - x, 2) + pow(ponto.y - y, 2));

    return d;
}
